plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1978,11 +1978,16 @@ So if getMinSpeed() returns null we fall back to getSpeed().
         final SerialDef serial = new SerialDef("pty", null, (short)0);
         devices.addDevice(serial);
 
+        /* Add a VirtIO channel for SystemVMs for communication and provisioning */
         if (vmTO.getType() != VirtualMachine.Type.User) {
-            final VirtioSerialDef vserial = new VirtioSerialDef(vmTO.getName(), null);
-            devices.addDevice(vserial);
+            devices.addDevice(new ChannelDef(vmTO.getName() + ".vport", ChannelDef.ChannelType.UNIX,
+                                             "/var/lib/libvirt/qemu/" + vmTO.getName() + ".agent"));
```

Comment:
This has always been hardcoded, it's just that I'm moving this part.

This is the SSVM socket which is already there. The socket patch code wants to find the .agent there. Not much I can do there.